loki/analyse/analyse_dataflow.py

```diff
@@ -150,9 +159,12 @@ def visit_MultiConditional(self, o, **kwargs):
         vset = set(v for v in FindVariables().visit(o.values) if not v in query_args)
 
         uses = self._symbols_from_expr(as_tuple(eset)) | self._symbols_from_expr(as_tuple(vset))
-        body, defines, uses = self._visit_body(o.bodies, live=live, uses=uses, **kwargs)
+        body = ()
+        defines = set()
+        for b in o.bodies:
+            _b, defines, uses = self._visit_body(b, live=live, uses=uses, defines=defines, **kwargs)
```

Comment:
Would it be possible to check the impact of this in [test_analyse_multiconditional](https://github.com/ecmwf-ifs/loki/blob/7fd6b8607967ee1d30281e36c63a86c54f1d054a/tests/test_analyse_dataflow.py#L413)? Not sure I'm understanding the original problem: did it re-order the cases?

Reply:
The bodies of the multiconditional should be arranged in a nested tuple, with the outer level corresponding to the number of branches for the multiconditional. When we run a dataflow analysis pass, it flattens the bodies of the multiconditional. Whilst this isn't a problem for the dataflow analysis, it's a problem if we then want to regenerate Fortran code from the IR (the node is updated).

Reply:
And yes, we can test the impact of this by querying the size of the inner and outer layer of `o.bodies`. I will amend the test accordingly.